lib/provider/data_types.go

```diff
@@ -105,3 +105,14 @@ type Snapshot struct {
 	// tags for the snapshot
 	SnapshotTags SnapshotTags `json:"tags"`
 }
+
+// AuthorizationRequest capture details of autorization to be made
+type AuthorizationRequest struct {
```

Comment:
Can we have different struct name may be `VolumeAuthRequest`, as per name purpose of this struct is not coming out.

Reply:
Done --> Changed to `VolumeAuthorization`